specification/resources/resource-manager/Microsoft.Resources/stable/2018-02-01/resources.json

```diff
@@ -0,0 +1,2955 @@
+            "pattern": "^[-\\w\\._\\(\\)]+$",
+            "minLength": 1,
+            "maxLength": 64
+          },
+          {
+            "name": "$top",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "description": "The number of results to return."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK - Return an array of deployment operations.",
+            "schema": {
+              "$ref": "#/definitions/DeploymentOperationsListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DeploymentExtendedFilter": {
+      "properties": {
+        "provisioningState": {
+          "type": "string",
+          "description": "The provisioning state."
+        }
+      },
+      "description": "Deployment filter."
+    },
+    "GenericResourceFilter": {
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "The resource type."
+        },
+        "tagname": {
+          "type": "string",
+          "description": "The tag name."
+        },
+        "tagvalue": {
+          "type": "string",
+          "description": "The tag value."
+        }
+      },
+      "description": "Resource filter."
+    },
+    "ResourceGroupFilter": {
+      "properties": {
+        "tagName": {
+          "type": "string",
+          "description": "The tag name."
+        },
+        "tagValue": {
+          "type": "string",
+          "description": "The tag value."
+        }
+      },
+      "description": "Resource group filter."
+    },
+    "TemplateLink": {
+      "properties": {
+        "uri": {
+          "type": "string",
+          "description": "The URI of the template to deploy."
+        },
+        "contentVersion": {
+          "type": "string",
+          "description": "If included, must match the ContentVersion in the template."
+        }
+      },
+      "required": [
+        "uri"
+      ],
+      "description": "Entity representing the reference to the template."
+    },
+    "ParametersLink": {
+      "properties": {
+        "uri": {
+          "type": "string",
+          "description": "The URI of the parameters file."
+        },
+        "contentVersion": {
+          "type": "string",
+          "description": "If included, must match the ContentVersion in the template."
+        }
+      },
+      "required": [
+        "uri"
+      ],
+      "description": "Entity representing the reference to the deployment paramaters."
+    },
+    "DeploymentProperties": {
+      "properties": {
+        "template": {
```

Comment:
`name`, `id` and `type` has to be `readonly` properties.

Reply:
for on error deployment type should be assignable by the user